test/python/compiler/test_assembler.py

```diff
@@ -632,9 +632,9 @@ def test_assemble_parametric(self):
         self.assertDictEqual(qobj_insts[2].parameters, {'duration': 25, 'amp': 1})
         self.assertDictEqual(qobj_insts[3].parameters,
                              {'duration': 150, 'sigma': 8, 'amp': 0.2, 'width': 140})
-        self.assertListEqual(
+        self.assertEqual(
             qobj.to_dict()['experiments'][0]['instructions'][0]['parameters']['amp'],
-            [0.0, 0.5])
+            0.0+0.5j)
```

Comment:
I think I understand why `to_dict` is now returning a complex type, rather than this list, but I don't see where (or if?) the list is generated before being sent to the provider. I'd expect a `NumpyEncoder` and `ComplexEncoder` like in `test_unitary` to get added to `assemble`. If we sent this complex type as is (though I thought json didn't support), the backend would not understand it.

Reply:
This is part of the decoupling between the qobj the class and what the ibmq provider sends on the wire and was what I labeled as a breaking change in the release notes. The encoder class should live in the qiskit-ibmq-provider. It is only for when we actually serialize to a json string (ie right before the provider sends the request out) to send to the iqx api that doing that conversion is needed.

For testing those converted classes are needed in a couple of places because we run the jsonschema validation which is meant for validating the wire format.